hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -612,10 +629,78 @@ protected Object loadFromSecondLevelCache(
 					persister.getEntityName()
 				);
 		}
-		
+
 		return entity;
 	}
 
+	private Object convertCacheReferenceEntryToEntity(
+			ReferenceCacheEntryImpl referenceCacheEntry,
+			Serializable entityId,
+			EntityPersister persister,
+			EventSource session,
+			EntityKey entityKey) {
+		final Object entity = referenceCacheEntry.getReference();
+
+		if ( entity == null ) {
+			throw new IllegalStateException(
+					"Reference cache entry contained null : " + entityId);
+		}
+		else {
+			makeEntityCircularReferenceSafe(referenceCacheEntry, entityId, session, entity, entityKey);
+			//PostLoad is needed for EJB3
+			EventListenerGroup<PostLoadEventListener> evenListenerGroup = getEvenListenerGroup(session);
+
+			if(!evenListenerGroup.isEmpty()) {
+				postLoad(session, evenListenerGroup.listeners(), entity, entityId, persister);
+			}
+			return entity;
+		}
```

Comment:
Not allocating an eventListernerGroup iterator if the listeners collection is empty